### PR TITLE
Avatar implementation simplification

### PR DESCRIPTION
**Pull request by simeonoff**
Closes # .  

Additional information related to this pull request:

### CODE REVIEW

File: src/avatar/avatar.component.ts [hunk 1]
```diff
@@ -23,49 +21,42 @@ export enum Size {
     selector: "igx-avatar",
     templateUrl: "avatar.component.html"
 })
-export class IgxAvatarComponent implements AfterViewInit, AfterContentChecked {
+export class IgxAvatarComponent implements AfterContentChecked {
```
Comment (rkaraivanov) [inline src/avatar/avatar.component.ts]: Does it really need to implement `AfterContentChecked`? The `getRole` method seems like something that should be called once and not every detection cycle

File: src/core/styles/components/avatar/_avatar-theme.scss [hunk 2]
```diff
@@ -2,29 +2,45 @@ $component-name: 'igx-avatar';
...
+    $small-size: 40px;
+    $medium-size: 64px;
+    $large-size: 88px;
 
     %igx-avatar-display {
-        display: flex;
-        justify-content: center;
-        align-items: center;
         position: relative;
         pointer-events: none;
```
Comment (bkulov) [inline src/core/styles/components/avatar/_avatar-theme.scss]: Please restore pointer-events as the user may need to implement some logic based on them. Use user-select set to none to disallow text/icon selection.

File: src/avatar/avatar.component.ts [hunk 3]
```diff
@@ -23,49 +21,42 @@ export enum Size {
...
-
     private _size: string;
     private _bgColor: string;
-    private _icon = "android";
-    private _isInitialsRepositionNeeded = false;
+    private _icon: string;
+    private _initials: string;
+    private _src: string;
+
+    @Input("roundShape") public roundShape = "false";
```
Comment (rkaraivanov) [inline src/avatar/avatar.component.ts]: IMO this should be a boolean instead of a simple string.

File: src/avatar/avatar.component.ts [hunk 4]
```diff
@@ -78,92 +69,61 @@ export class IgxAvatarComponent implements AfterViewInit, AfterContentChecked {
...
 
-    public get srcImage() {
-        return this.image ? this.image.nativeElement.src : "";
-    }
-
-    public set srcImage(value: string) {
-        this.image.nativeElement.src = value;
-    }
-
     get isRounded(): boolean {
```
Comment (rkaraivanov) [inline src/avatar/avatar.component.ts]: If the `roundShape` property is boolean this is redundant.

File: src/avatar/avatar.component.ts [hunk 5]
```diff
@@ -23,49 +21,42 @@ export enum Size {
...
     get size(): string {
-        if (this._isInitialsRepositionNeeded) {
-            this.repositionInitials();
-        }
-
         return this._size === undefined ? "small" : this._size;
     }
 
     set size(value: string) {
         const sizeType = this.sizeEnum[value.toUpperCase()];
```
Comment (rkaraivanov) [inline src/avatar/avatar.component.ts]: Typescript supports string enumerations. Initialize the `Size` enumeration with the strings for the size and this check is kind of obsolete.

File: src/avatar/avatar.component.ts [hunk 6]
```diff
@@ -23,49 +21,42 @@ export enum Size {
     selector: "igx-avatar",
     templateUrl: "avatar.component.html"
 })
-export class IgxAvatarComponent implements AfterViewInit, AfterContentChecked {
+export class IgxAvatarComponent implements OnInit {
     @ViewChild("image") public image: ElementRef;
```
Comment (rkaraivanov) [inline src/avatar/avatar.component.ts]: Code formatting.
Put the property below the decorator and space them a bit.

File: src/avatar/avatar.component.ts [hunk 7]
```diff
@@ -23,49 +21,42 @@ export enum Size {
...
     @ViewChild("initialsImage") public initialsImage: ElementRef;
-    @Input() public initials: string;
-    @Input() public src: string;
-    @Input("roundShape") public roundShape = "false";
-    @Input() public color = "white";
+    @ViewChild("imageTemplate", { read: TemplateRef }) protected imageTemplate: TemplateRef<any>;
```
Comment (rkaraivanov) [inline src/avatar/avatar.component.ts]: Same as above.

File: src/avatar/avatar.component.ts [hunk 8]
```diff
@@ -23,49 +21,42 @@ export enum Size {
...
+    @ViewChild("initialsTemplate", { read: TemplateRef }) protected initialsTemplate: TemplateRef<any>;
+    @ViewChild("iconTemplate", { read: TemplateRef }) protected iconTemplate: TemplateRef<any>;
 
     @HostBinding("attr.aria-label") public ariaLabel = "avatar";
     @HostBinding("attr.role") public role = "img";
```
Comment (rkaraivanov) [inline src/avatar/avatar.component.ts]: Same as above.

**Pull request merged**